src/servers/src/http/jaeger.rs

```diff
@@ -0,0 +1,1244 @@
+fn get_operations_sql(
+    db: &str,
+    trace_table_name: &str,
+    service_name: &str,
+    span_kind: Option<&str>,
+) -> String {
+    let mut query = format!(
+        "SELECT span_name, span_kind FROM {}.{} WHERE service_name = '{}'",
+        db, trace_table_name, service_name
+    );
+    if let Some(span_kind) = span_kind {
+        query = format!(
+            "{} AND span_kind = '{}'",
+            query,
+            normalize_span_kind(span_kind)
+        );
+    }
+    query
+}
+
+fn query_trace_sql(trace_table_name: &str, query_params: QueryTraceParams) -> String {
+    let mut query = format!(
+        "SELECT trace_id, timestamp, duration_nano, service_name, span_name, span_id, span_attributes FROM {}.{} WHERE service_name = '{}'",
+        query_params.db, trace_table_name, query_params.service_name
+    );
+
+    if let Some(operation_name) = query_params.operation_name {
+        query = format!("{} AND span_name = '{}'", query, operation_name);
+    }
+
+    if let Some(start_time) = query_params.start_time {
+        query = format!("{} AND timestamp >= {}", query, start_time);
+    }
+
+    if let Some(end_time) = query_params.end_time {
+        query = format!("{} AND timestamp <= {}", query, end_time);
+    }
+
+    if let Some(min_duration) = query_params.min_duration {
+        query = format!("{} AND duration_nano >= {}", query, min_duration);
+    }
+
+    if let Some(max_duration) = query_params.max_duration {
+        query = format!("{} AND duration_nano <= {}", query, max_duration);
+    }
+
+    // Use json functions to filter traces by tags.
+    if let Some(tags) = query_params.tags {
+        for (key, value) in tags.iter() {
+            if let JsonValue::String(value) = value {
+                query = format!(
+                    "{} AND json_get_string(span_attributes, '[\"{}\"]') = '{}'",
+                    query, key, value
+                );
+            }
+            if let JsonValue::Number(value) = value {
+                if value.is_i64() {
+                    query = format!(
+                        "{} AND json_get_int(span_attributes, '[\"{}\"]') = {}",
+                        query, key, value
+                    );
+                }
+                if value.is_f64() {
+                    query = format!(
+                        "{} AND json_get_float(span_attributes, '[\"{}\"]') = {}",
+                        query, key, value
+                    );
+                }
+            }
+            if let JsonValue::Bool(value) = value {
+                query = format!(
+                    "{} AND json_get_bool(span_attributes, '[\"{}\"]') = {}",
+                    query, key, value
+                );
+            }
+        }
+    }
+
+    if let Some(limit) = query_params.limit {
+        query = format!("{} LIMIT {}", query, limit);
+    }
+
+    query
+}
+
+fn normalize_span_kind(span_kind: &str) -> String {
```

Comment:
What about moving thins function into `SpanKind`?

Reply:
`span_kind` is a `String` type and I think it may be unnecessary to define the the new type and implement the above method.